concrete/src/Mail/MailServiceProvider.php

```diff
@@ -2,18 +2,25 @@
 namespace Concrete\Core\Mail;
 
 use Concrete\Core\Foundation\Service\Provider as ServiceProvider;
+use Concrete\Core\Mail\Transport\Smtp as SmtpTransport;
 
 class MailServiceProvider extends ServiceProvider
 {
     public function register()
     {
-        $register = array(
+        $app = $this->app;
+
+        $register = [
             'helper/mail' => '\Concrete\Core\Mail\Service',
             'mail' => '\Concrete\Core\Mail\Service',
-        );
+        ];
 
         foreach ($register as $key => $value) {
-            $this->app->bind($key, $value);
+            $app->bind($key, $value);
         }
+
+        $this->app->singleton(SmtpTransport::class, function () use ($app) {
```

Comment:
We need a way to:
1. Use a single transport instance (so it's a singleton)
2. Keep track of the number of the email we have sent and close/reopen yhe connection exery x emails
3. Reset the connection in case of errors, in case we've lost it